src/embedding.rs

```diff
@@ -369,13 +537,27 @@ where
             self.dimension, self.number_of_entities
         );
 
-        let result = M::init_with_hashes(
-            self.number_of_entities,
-            self.dimension,
-            self.fixed_random_value,
-            self.sparse_matrix_reader.clone(),
-        );
-
+        let result = match self.init_method {
+            InitMethod::Random => M::init_with_hashes(
+                self.number_of_entities,
+                self.dimension,
+                self.fixed_random_value,
+                self.sparse_matrix_reader.clone(),
+            ),
+            InitMethod::EvecSmallest => M::init_with_evec( // Is there a more compact solution?
+                self.number_of_entities,
+                self.dimension,
+                self.sparse_matrix_reader.clone(),
```

Comment:
I can see that you allow in method interface to use different TruncatedOrder. But isnt that the case that only one order make sense for laplacian? If that the case lets simplify the interface and inline proper value inside.

Reply:
We've experimented with both the smallest and the largest eigenvalues, and it seems like both can be used under the right circumstances, that's why I added both.